flink-container/src/main/java/org/apache/flink/container/entrypoint/ClassPathJobGraphRetriever.java

```diff
@@ -73,27 +76,31 @@
 			@Nonnull JobID jobId,
 			@Nonnull SavepointRestoreSettings savepointRestoreSettings,
 			@Nonnull String[] programArguments,
-			@Nullable String jobClassName) {
-		this(jobId, savepointRestoreSettings, programArguments, jobClassName, JarsOnClassPath.INSTANCE);
+			@Nullable String jobEntryPointName,
+			@Nullable String jobPythonArtifacts) {
+		this(jobId, savepointRestoreSettings, programArguments, jobEntryPointName, jobPythonArtifacts, JarsOnClassPath.INSTANCE);
```

Comment:
Why do we need to touch the `ClassPathJobGraphRetriever` at all? This class should not need to know about Python. Otherwise we need to add special casing for all supported languages in the future. This does not seem right.